src/west/app/project.py

```diff
@@ -1071,6 +1072,67 @@ def do_add_parser(self, parser_adder):
                            help='''cached repositories are in the same relative
                            paths as the workspace being updated. This cache has
                            lower priority (Prio 1).''')
+        group.add_argument('--auto-cache',
+                           help='''automatically setup local cache repositories
+                           in a flat folder hierarchy, but with an additional
+                           subfolder (hashed name) for different remote URLs.
+                           Each local cache repository is automatically cloned
+                           on first usage and synced on subsequent clones.
+                           This cache has the lowest priority (Prio 2).''')
+
+        parser.epilog = textwrap.dedent('''\
+        CACHING
+        -------
+        Projects are typically initialized by fetching from remote URLs. However,
+        they can also be cloned from local caches on the file system. Three types of
+        caches are supported, and they may be used together. They are searched in the
+        following order of priority:
+          - Priority 0: --name-cache
+          - Priority 1: --path-cache
+          - Priority 2: --auto-cache
+
+        Local caches are only used during the initial setup of your workspace.
```

Comment:
@thorsten-klein this is not quite true in the case of auto-cache, is it? you fetch into the auto-cache from the remote and then fetch from the auto-cache into the actual project folder?